### ISSUE TITLE
Error while highlighting Perl code

### ISSUE THREAD

**Issue by alepulver** [sonar-perl/sonar-perl#103]
The following code causes the parser to generate an empty token, and the highlighter fails:

```
warn <<'WARN';
WAR_ING:
WARN
```

Stacktrace:

```
Caused by: java.lang.IllegalArgumentException: Unable to highlight file lib/MyFile.pm
	at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:97)
	at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:37)
	at com.github.sonarperl.highlighter.PerlHighlighter.highlight(PerlHighlighter.java:66)
	... 40 more
Caused by: java.lang.IllegalArgumentException: Start pointer [line=2, lineOffset=0] should be before end pointer [line=2, lineOffset=0]
	at org.sonar.api.internal.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at org.sonar.api.batch.fs.internal.DefaultInputFile.newRangeValidPointers(DefaultInputFile.java:325)
	at org.sonar.api.batch.fs.internal.DefaultInputFile.newRange(DefaultInputFile.java:263)
	at org.sonar.api.batch.sensor.highlighting.internal.DefaultHighlighting.highlight(DefaultHighlighting.java:95)
	... 42 more
```

Is it possible to disable this scanning phase?

**Comment by rkrisztian** [comment]
I'm getting the same problem with similar patterns:

```perl
		print <<EOF;

EOF
```

```perl
    print $file <<"EOF";
# Comments line 1
# Comment line 2
EOF
```

```perl
                print <<"EOF";
            "$var1" -> "$var2";
EOF
```

```perl
     die <<"ERROR" if @errors;
 ERROR: Some problem with '$var1':
  @errors
 ERROR
```

(In the code above, the error happens at line `ERROR: (...)`.)

**Comment by tomaso** [comment]
I am getting the same error. Interestingly it goes away if I move the semicolon to the end of the heredoc:
`		print <<EOF

EOF
;`

**Comment by vitstradal** [comment]
The problem with `'EOL'` in parsing is IMHO  [here](https://github.com/sonar-perl/sonar-perl/blob/master/perl-frontend/src/main/java/com/github/sonarperl/lexer/HeredocChannel.java#L47):
```
lookFor = heredocDelimiter.getValue().substring(1, heredocDelimiter.getValue().length() - 2);
```
Where HeredocChannel tryes remove apostrofs (or quotes) from `'EOL`` (to get simly `EOL`), but imho gets `EO`, because of [substring(startindex, endindex)](https://www.javatpoint.com/java-string-substring).

**Comment by rkrisztian** [comment]
It should be possible to fix the plugin so that in case of parsing errors, the analysis would just continue without highlighting the line:

https://community.sonarsource.com/t/highlighting-errors-should-not-be-fatal/24833

(For me this was not so obvious, sorry.)

**Comment by tomaso** [comment]
@rkrisztian actually this is already implemented in https://github.com/sonar-perl/sonar-perl/pull/104/commits/d1eae489beb90fa07f8191352fb7d752a66e5f6d but the latest release of sonar-perl plugin (0.4.6.) doesn't contain this commit. 

I have just checked that a scan with this fix produces an exception for our heredoc but is successful overall.

Could I ask someone (@otrosien ? :) ) to build a new release with that commit?

**Comment by otrosien** [comment]
Hey @tomaso, hey @rkrisztian, sure, let me try. I haven't released anything for quite some time though, not sure if it still works.

**Comment by otrosien** [comment]
Ok, publishing the docker image doesn't work anymore. I will need a few days to figure out and fix it.

**Comment by otrosien** [comment]
I updated master branch to be compatible with the current LTS, I will test some more and release in the next days. If you want to play around, just build the plugin jar locally and install it into your SonarQube.

**Comment by otrosien** [comment]
The "crashing" on parse errors should be fixed in the latest release (0.5.2). Please check and re-open if you still see issues.
